account-kit/smart-contracts/src/ma-v2/modules/time-range-module/module.ts

```diff
@@ -0,0 +1,82 @@
+import { encodeAbiParameters, type Address, type Hex } from "viem";
+
+import { timeRangeModuleAbi } from "./abis/timeRangeModuleAbi.js";
+
+const addresses = {
+  default: "0x00", // TO DO: replace with deployed module address?
+} as Record<number | "default", Address>;
+
+const meta = {
+  name: "TimeRangeModule",
+  version: "alpha.1",
+  addresses,
+};
+
+// Todo: some unified type for ERC-6900 v0.8 modules. I couldn't figure out how to parameterize the class itself over the abi type parameters for onInstall and onUninstall.
+export const TimeRangeModule = {
+  meta,
+  abi: timeRangeModuleAbi,
+  encodeOnInstallData: (args: {
+    entityId: number;
+    validUntil: number;
+    validAfter: number;
+  }): Hex => {
+    const { entityId, validUntil, validAfter } = args;
+
+    return encodeAbiParameters(
+      [
+        {
+          type: "uint32",
+          value: entityId,
```

Comment:
As in other PR: the `value` field doesn't do anything here, should be just `type` and optionally `name`.